File: 1433.check-if-a-string-can-break-another-string.py
#
# @lc app=leetcode id=1433 lang=python3
#
# [1433] Check If a String Can Break Another String
#

# @lc code=start
class Solution:
    # This question is pretty simple
    # Sort s1 and s2
    # Check if s1[i] <= s2[i] or s2[i] <= s1[i] for all i
    def checkIfCanBreak(self, s1: str, s2: str) -> bool:
        s1 = ''.join(sorted([c for c in s1]))
        s2 = ''.join(sorted([c for c in s2]))
        return all([s1[i] >= s2[i] for i in range(len(s1))]) or all([s1[i] <= s2[i] for i in range(len(s1))])
        
# @lc code=end

File: 1433.check-if-a-string-can-break-another-string.cs
/*
 * @lc app=leetcode id=1433 lang=csharp
 *
 * [1433] Check If a String Can Break Another String
 */

// @lc code=start
public class Solution {
    public bool CheckIfCanBreak(string s1, string s2) {
        int n = s1.Length;
        char[] s1Arr = s1.ToCharArray();
        char[] s2Arr = s2.ToCharArray();

        Array.Sort(s1Arr);
        Array.Sort(s2Arr);
        int count1 = 0;
        int count2 = 0;
        for (int i = 0; i < n; i++)
        {
            if (s1Arr[i] > s2Arr[i])
            {
                count1 += 1;
            }
            else if (s1Arr[i] < s2Arr[i])
            {
                count2 += 1;
            }
        }
        return (count1 == 0 || count2 == 0);
    }
}
// @lc code=end

